feat(react-scheduler): redesign AppointmentTooltip (#2474)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/appointment-tooltip.md

headerComponent | ComponentType&lt;[AppointmentTooltip.HeaderProps](#appointmenttooltipheaderprops)&gt; | | A component that renders the header.
contentComponent | ComponentType&lt;[AppointmentTooltip.ContentProps](#appointmenttooltipcontentprops)&gt; | | A component that renders the tooltip content.
commandButtonComponent | ComponentType&lt;[AppointmentTooltip.CommandButtonProps](#appointmenttooltipcommandbuttonprops)&gt; | | A component that renders a command button.

> recurringIconComponent | ComponentType&lt;object&gt; | | A component that renders an icon for recurring appointments inside the AppointmentTooltip.

Reviewer: ```suggestion
recurringIconComponent | ComponentType&lt;object&gt; | | A component that renders an icon that indicates a recurring appointment.
```

> commandButtonIds | Array&lt;string&gt; | A command button's identifier list.

Reviewer 2: "The command button", if you are talking about a specific button.

formatDate | [FormatterFn](scheduler.md#formatterfn) | A function that formats dates according to the locale.
onOpenButtonClick? | () => void | An event raised when the Open button is clicked. The event handler should open the appointment form.
onDeleteButtonClick? | () => void | An event raised when the Delete button is clicked. The event handler should delete an appointment.
appointmentMeta? | [AppointmentMeta](#appointmentmeta) | The appointment's displayed metadata.
visible? | boolean | Specifies the tooltip's visibility.
onHide? | () => void | An event that hides the tooltip.
headerComponent | ComponentType&lt;[AppointmentTooltip.HeaderProps](#appointmenttooltipheaderprops)&gt; | A component that renders the tooltip header.
contentComponent | ComponentType&lt;[AppointmentTooltip.ContentProps](#appointmenttooltipcontentprops)&gt; | A component that renders the tooltip content.
commandButtonComponent | ComponentType&lt;[AppointmentTooltip.CommandButtonProps](#appointmenttooltipcommandbuttonprops)&gt; | A component that renders a command button.

> recurringIconComponent | ComponentType&lt;object&gt; | A component that renders an icon for recurring appointments inside the AppointmentTooltip.

Reviewer: ```suggestion
recurringIconComponent | ComponentType&lt;object&gt; | A component that renders an icon that indicates a recurring appointment.
```

### AppointmentTooltip.HeaderProps

Describes properties passed to a component that renders the tooltip header.

Field | Type | Description
------|------|------------
showOpenButton? | boolean | Specifies the Open button's visibility.
showCloseButton? | boolean | Specifies the Close button's visibility.
showDeleteButton? | boolean | Specifies the Delete button's visibility.
appointmentData? | [AppointmentModel](./scheduler.md#appointmentmodel) | The appointment's displayed metadata.
onOpenButtonClick? | () => void | An event raised when the Open button is clicked. The event handler should open the appointment form.
onDeleteButtonClick? | () => void | An event raised when the Delete button is clicked. The event handler should delete an appointment.

> onHide? | () => void | An event that hides the tooltip.

Reviewer: ```suggestion
onHide? | () => void | An event raised when the tooltip hides.
```

commandButtonComponent | ComponentType&lt;[AppointmentTooltip.CommandButtonProps](#appointmenttooltipcommandbuttonprops)&gt; | A component that renders a command button.

> children? | ReactNode | A React node used to render the additional part of the tooltip header.

Reviewer: ```suggestion
children? | ReactNode | A React node that renders custom elements in the tooltip header.
```

Field | Type | Description
------|------|------------
appointmentData? | [AppointmentModel](./scheduler.md#appointmentmodel) | The appointment's displayed metadata.
formatDate | [FormatterFn](scheduler.md#formatterfn) | A function that formats dates according to the locale.

> recurringIconComponent | ComponentType&lt;object&gt; | A component that renders an icon for recurring appointments inside the AppointmentTooltip.

Reviewer: ```suggestion
recurringIconComponent | ComponentType&lt;object&gt; | A component that renders an icon that indicates a recurring appointment.
```

> children? | ReactNode | A React node used to render the additional part of the tooltip content.

Reviewer: ```suggestion
children? | ReactNode | A React node that renders custom elements in the tooltip.
```
